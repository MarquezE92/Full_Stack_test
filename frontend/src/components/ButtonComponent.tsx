import React from 'react';
import { ButtonProps } from '../types';
const ButtonComponent: React.FC<ButtonProps>= ({label, isOn, handleClick})=>{
    return(
        <button className={isOn? '' : 'inactiveButton'} onClick={isOn?handleClick : ()=>{}}>{label}</button>
    )
}

export default ButtonComponent