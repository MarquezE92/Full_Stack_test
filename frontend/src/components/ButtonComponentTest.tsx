/* import React from 'react';
const ButtonComponentForTest=({label, isOn, handleClick})=>{
    return(
        <button className={isOn? '' : 'inactiveButton'} onClick={isOn?handleClick : ()=>{}}>{label}</button>
    )
}

export {ButtonComponentForTest} */