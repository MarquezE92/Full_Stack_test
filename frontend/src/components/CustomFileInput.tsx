import React, { useRef } from 'react';
import { customInputProps } from '../types';

const CustomFileInput: React.FC<customInputProps> = ({ onChange, isOn }) => {
    const ref = useRef<HTMLInputElement>(null);
  
    const handleClick = () => {
      ref.current?.click();
    };
  
    return (
      <div className={isOn ? 'custom-file-input-selected' :'custom-file-input'} onClick={handleClick}>
        <input ref={ref} type='file' accept='.csv' onChange={onChange} style={{ display: 'none' }} />
       {isOn ? 'File Selected' : 'Select CSV File'}
      </div>
    );
  };

  export default CustomFileInput