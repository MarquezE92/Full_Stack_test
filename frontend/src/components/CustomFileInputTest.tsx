import React, { useRef } from 'react';

const CustomFileInputTest = ({ onChange, isOn }) => {
    const ref = useRef(null);
  
    const handleClick = () => {
      ref.current?.click();
    };
  
    return (
      <div className={isOn ? 'custom-file-input-selected' :'custom-file-input'} onClick={handleClick}>
        <input ref={ref} type='file' accept='.csv' onChange={onChange} style={{ display: 'none' }} />
       {isOn ? 'File Selected' : 'Select CSV File'}
      </div>
    );
  };

  export default CustomFileInputTest