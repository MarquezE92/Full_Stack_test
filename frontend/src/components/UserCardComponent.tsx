import React from 'react';
import { CardProps } from '../types';

const UserCardComponent: React.FC<CardProps> = ({name, city='?', country='?', favorite_sport='?'})=>{
    return(
        <div className='card'>
            <h2>{name}</h2>
            <h3>City: {city}</h3>
            <h3>Country: {country}</h3>
            <h3>Favorite sport: {favorite_sport}</h3>
        </div>
    )
}

export default UserCardComponent