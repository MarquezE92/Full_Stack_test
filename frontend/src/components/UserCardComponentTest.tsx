/* import React from 'react';

const UserCardComponentTest = ({name, city='?', country='?', favorite_sport='?'})=>{
    return(
        <div className='card'>
            <h2>{name}</h2>
            <h3>City: {city}</h3>
            <h3>Country: {country}</h3>
            <h3>Favorite sport: {favorite_sport}</h3>
        </div>
    )
}

export default UserCardComponentTest */