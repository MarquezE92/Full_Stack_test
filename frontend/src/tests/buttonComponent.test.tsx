import React from 'react';
import TestRenderer from 'react-test-renderer'
import {ButtonComponentForTest} from '../components/ButtonComponentTest';

describe('ButtonComponent testing', ()=> {
    test('ButtonComponent renders correctly', () => {
  // Arrange
  const mockData = {
    label: 'Upload file',
    isOn: true,
    handleClick: jest.fn(), // Mock handleClick function
  };

  
  const treeLine = TestRenderer.create(<ButtonComponentForTest {...mockData} />).toJSON()
  expect(treeLine).toMatchSnapshot()
});

})
