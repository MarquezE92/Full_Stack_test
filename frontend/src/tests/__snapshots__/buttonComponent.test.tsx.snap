// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonComponent renders correctly 1`] = `
<button
  className=""
  onClick={[MockFunction]}
>
  Upload file
</button>
`;

exports[`ButtonComponent testing ButtonComponent renders correctly 1`] = `
<button
  className=""
  onClick={[MockFunction]}
>
  Upload file
</button>
`;
